frontend/apps/pwa/src/features/former/validation.ts

```diff
@@ -0,0 +1,135 @@
+import { FieldPath } from 'react-hook-form';
+import { FieldKind } from 'core-api-client';
+
+import { Form, FormField, ValidationForm } from '../../reducers/Former/types';
+
+const validationConstants = {
+  name: {
+    minLength: 3,
+    maxLength: 128,
+    pattern: /[A-Za-z0-9]/,
+  },
+  options: {
+    min: 2,
+    max: 60,
+    name: {
+      minLength: 2,
+      maxLength: 128,
+    },
+  },
+  fields: {
+    min: 1,
+    max: 100,
+  },
+};
+
+export const registeredValidation = {
+  name: {
+    minLength: {
+      value: validationConstants.name.minLength,
+      message: `Form name needs to be at least ${validationConstants.name.minLength} characters long`,
+    },
+    maxLength: {
+      value: validationConstants.name.maxLength,
+      message: `Form name can be at most ${validationConstants.name.maxLength} characters long`,
+    },
+    pattern: {
+      value: validationConstants.name.pattern,
+      message: 'Form name contains invalid characters',
+    },
+    required: { value: true, message: 'Form name is a required field' },
+  },
+  selectedField: {
+    name: {
+      minLength: {
+        value: validationConstants.name.minLength,
+        message: `Field name needs to be at least ${validationConstants.name.minLength} characters long`,
+      },
+      maxLength: {
+        value: validationConstants.name.maxLength,
+        message: `Field name can be at most ${validationConstants.name.maxLength} characters long`,
+      },
+      required: { value: true, message: 'Field name is required' },
+    },
+    fieldType: {
+      reference: {
+        databaseId: {
+          required: { value: true, message: 'Data base is required' },
+        },
+        formId: {
+          required: { value: true, message: 'Form is required' },
+        },
+      },
+      select: {
+        option: {
+          name: {
+            minLength: {
+              value: validationConstants.name.minLength,
+              message: `Option name needs to be at least ${validationConstants.name.minLength} characters long`,
+            },
+            maxLength: {
+              value: validationConstants.name.maxLength,
+              message: `Option name can be at most ${validationConstants.name.maxLength} characters long`,
+            },
+            required: { value: true, message: 'Option name is required' },
+          },
+        },
+      },
+    },
+  },
+};
+
+type CustomError = {
+  field: FieldPath<ValidationForm>;
+  message: string;
+};
+
+export const customValidation = {
```

Comment:
Could this not be part of react form hooks validation? I haven't given any thought to how, but it would be nice if it handled everything.

Maybe we have two instances of react form hooks. One validates the form and one validates the current field.

Reply:
I didn't think so, because register only works on input and select elements, and we have things that need to be validated that are not bound to one of those, like amount of fields, amount of options, and such.
We could refactor the whole form and validation to work with the control mechanism from react-hook-form, but that would not be a small change.

Reply:
Yeah agreed. I think when we build out the native app we should think with react form hooks from the start and try to use it properly. Another one to look at for this custom validation would be io-ts, but I think this suffices for now (I also don't think that would catch things like number of fields).